ut/columns_ut.cpp

```diff
@@ -199,6 +210,29 @@ TEST(ColumnsCase, Date32_Int32_interface) {
     ASSERT_EQ(col1->RawAt(2), -1234);
 }
 
+TEST(ColumnsCase, Date32_construct_from_rvalue_data) {
+    auto const expected = MakeNumbers<int32_t>();
+
+    auto data = expected;
+    auto col1 = std::make_shared<ColumnDate32>(std::move(data));
+
+    ASSERT_EQ(col1->Size(), expected.size());
+    for (size_t i = 0; i < expected.size(); ++i) {
+        ASSERT_EQ(col1->RawAt(i), expected[i]);
+    }
+}
+
+TEST(ColumnsCase, DateTime_construct_from_rvalue_data) {
+    auto const expected = MakeNumbers<uint32_t>();
+
+    auto data = expected;
+    auto col1 = std::make_shared<ColumnDateTime>(std::move(data));
+
+    ASSERT_EQ(col1->Size(), expected.size());
+    for (size_t i = 0; i < expected.size(); ++i) {
+        ASSERT_EQ(col1->At(i), static_cast<std::time_t>(expected[i]));
+    }
+}
```

Comment:
you can use a single check instead of  for-loop here:
```C++
EXPECT_TRUE(CompareRecursive(*col1, expected));
```
Takes care of size checks, value checks, and forms a pretty explanation message if comparison fails.